File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/event/rollcall/RollCallEntity.java
package com.github.dedis.popstellar.repository.database.event.rollcall;

import androidx.annotation.NonNull;
import androidx.room.*;

import com.github.dedis.popstellar.model.Immutable;
import com.github.dedis.popstellar.model.objects.RollCall;

@Entity(tableName = "rollcalls")
@Immutable
public class RollCallEntity {

  @PrimaryKey
  @ColumnInfo(name = "rollcall_id")
  @NonNull
  private final String rollcallId;

  @ColumnInfo(name = "lao_id", index = true)
  @NonNull
  private final String laoId;

  @ColumnInfo(name = "rollcall")
  @NonNull
  private final RollCall rollCall;

  public RollCallEntity(
      @NonNull String rollcallId, @NonNull String laoId, @NonNull RollCall rollCall) {
    this.rollcallId = rollcallId;
    this.laoId = laoId;
    this.rollCall = rollCall;
  }

  @Ignore
  public RollCallEntity(@NonNull String laoId, @NonNull RollCall rollCall) {
    this(rollCall.getPersistentId(), laoId, rollCall);
  }

  @NonNull
  public String getRollcallId() {
    return rollcallId;
  }

  @NonNull
  public String getLaoId() {
    return laoId;
  }

  @NonNull
  public RollCall getRollCall() {
    return rollCall;
  }
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/event/rollcall/RollCallEntity.kt
package com.github.dedis.popstellar.repository.database.event.rollcall

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey
import com.github.dedis.popstellar.model.Immutable
import com.github.dedis.popstellar.model.objects.RollCall

@Entity(tableName = "rollcalls")
@Immutable
class RollCallEntity(
    @field:ColumnInfo(name = "rollcall_id") @field:PrimaryKey val rollcallId: String,
    @field:ColumnInfo(name = "lao_id", index = true) val laoId: String,
    @field:ColumnInfo(name = "rollcall") val rollCall: RollCall
) {

  @Ignore
  constructor(laoId: String, rollCall: RollCall) : this(rollCall.persistentId, laoId, rollCall)
}
